lib/nerves_hub_link/pub_sub.ex

```diff
@@ -0,0 +1,75 @@
+defmodule NervesHubLink.PubSub do
+  @moduledoc """
+  PubSub mechanism on top of basic Elixir Registry primitives.
+
+  Wrapped up in a module for convenience and convenient correct use in
+  libraries that extend NervesHubLink via this mechanism.
```

Comment:
Do we want to allow external use of this PubSub? I was imagining this as one-way. Link is the only publisher, and everything external are subscribers. For the external extensions to interact with Link, it requires functions to be defined and called directly

Reply:
We could make publish private since we changed pushing messages up hubwards to be a GenServer.call. It wasn't, earlier.

I want extensions to have access to a subscribe function at least.